File: Android/samples/demo/demo/src/main/java/com/didi/aoe/examples/demo/ShowcaseApp.java
package com.didi.aoe.examples.demo;

import android.app.Application;
import android.content.Context;

/**
 * @author noctis
 */
public class ShowcaseApp extends Application {
    private static Context mContext;

    public static Context getContext() {
        return mContext;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();
    }
}

File: Android/examples/demo/src/main/java/com/didi/aoe/examples/demo/ShowcaseApp.kt
package com.didi.aoe.examples.demo

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import com.didi.aoe.library.service.AoeDataProvider
import com.didi.aoe.library.service.AoeService

/**
 * @author noctis
 */
class ShowcaseApp : Application() {

    override fun onCreate() {
        super.onCreate()
        context = applicationContext

        AoeService.init(applicationContext)

    }

    companion object {
        @SuppressLint("StaticFieldLeak")
        var context: Context? = null
            private set
    }

}